runtime/gc_glue_java/FlattenedArrayObjectScanner.hpp

```diff
@@ -0,0 +1,258 @@
+		fomrobject_t *endPtr = limitPtr;
+		if (!GC_ObjectScanner::isIndexableObjectNoSplit(flags) && (splitAmount != 0)) {
+			Assert_MM_unreachable();
+			endPtr = (fomrobject_t *)((uintptr_t)scanPtr + splitAmount*elementStride);
+			if (endPtr > limitPtr) {
+				endPtr = limitPtr;
+			}
+		}
+		uintptr_t *instanceDescription = elementClass->instanceDescription;
+#if defined(OMR_GC_LEAF_BITS)
+		uintptr_t *leafDescription = elementClass->instanceLeafDescription;
+#endif /* defined(OMR_GC_LEAF_BITS) */
+
+#if defined(OMR_GC_LEAF_BITS)
+		new(objectScanner) GC_FlattenedArrayObjectScanner(env, objectPtr, basePtr, limitPtr, scanPtr, endPtr, elementSize, elementStride, instanceDescription, leafDescription, flags);
+#else /* defined(OMR_GC_LEAF_BITS) */
+		new(objectScanner) GC_FlattenedArrayObjectScanner(env, objectPtr, basePtr, limitPtr, scanPtr, endPtr, elementSize, elementStride, instanceDescription, flags);
+#endif /* defined(OMR_GC_LEAF_BITS) */
+
+		objectScanner->initialize(env);
+		if (0 != startIndex) {
+			objectScanner->clearHeadObjectScanner();
+		}
+		return objectScanner;
+	}
+
+	MMINLINE uintptr_t getBytesRemaining() { return sizeof(fomrobject_t) * (_endPtr - _scanPtr); }
+
+	/**
+	 * @param env The scanning thread environment
+	 * @param allocSpace pointer to space within which the scanner should be instantiated (in-place)
+	 * @param splitAmount The maximum number of array elements to include
+	 * @return Pointer to split scanner in allocSpace
+	 */
+	GC_IndexableObjectScanner *
+	splitTo(MM_EnvironmentBase *env, void *allocSpace, uintptr_t splitAmount)
+	{
+		Assert_MM_unimplemented();
+	}
+
+	/**
+	 * Return base pointer and slot bit map for next block of contiguous slots to be scanned. The
+	 * base pointer must be fomrobject_t-aligned. Bits in the bit map are scanned in order of
+	 * increasing significance, and the least significant bit maps to the slot at the returned
+	 * base pointer.
+	 *
+	 * @param[out] scanMap the bit map for the slots contiguous with the returned base pointer
+	 * @param[out] hasNextSlotMap set this to true if this method should be called again, false if this map is known to be last
+	 * @return a pointer to the first slot mapped by the least significant bit of the map, or NULL if no more slots
+	 */
+	virtual fomrobject_t *
+	getNextSlotMap(uintptr_t *slotMap, bool *hasNextSlotMap)
+	{
+		fomrobject_t *result = GC_HeadlessMixedObjectScanner::getNextSlotMap(slotMap, hasNextSlotMap);
+		/* Ignore hasNextSlotMap from HeadLess, we want to always report that there is another element */
+		*hasNextSlotMap = true;
+		if (result == NULL) {
+			/* No more slots in the current element, get the next element of the array */
+			result = _indexableScanner.nextIndexableElement();
+			if (result == NULL) {
+				/* There are no elements in the array */
+				*hasNextSlotMap = false;
+			} else {
+				_mapPtr = result;
+				_endPtr = (fomrobject_t *)((uint8_t*)_mapPtr + _elementSizeWithoutPadding);
```

Comment:
Is there a reason to cast to `(uint8_t *)`? There is cast to `(uintptr_t)` used everywhere. However if you want to keep `(uint8_t *)` - formatting: please add space between `uint8_t` and `*`

Reply:
I cast to `uintptr_t` instead

Reply:
Fixed